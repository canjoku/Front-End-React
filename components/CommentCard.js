import { Component } from 'react'

class CommentCard extends Component {

  render () {
    const {comment} = this.props
    return (
      <div>
        <p>{comment.body}</p>
        <p>{comment.name}</p>
        <img src={comment.picture}/>
      </div>
    )
  }
}

export default CommentCard