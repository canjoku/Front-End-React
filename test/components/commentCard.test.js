import CommentCard from '../../components/CommentCard'

test('<CommentCard /> renders correctly', () => {
  const comment = {
    "body": "body of comment",
    "name" : "tony",
    "picture": "path to pic"
  }

  const component = shallow(<CommentCard comment={comment}/>)
  expect(component).toMatchSnapshot()
})
