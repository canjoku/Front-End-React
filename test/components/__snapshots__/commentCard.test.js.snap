// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentCard /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentCard
    comment={
      Object {
        "body": "body of comment",
        "name": "tony",
        "picture": "path to pic",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          body of comment
        </p>,
        <p>
          tony
        </p>,
        <img
          src="path to pic"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "body of comment",
        },
        "ref": null,
        "rendered": "body of comment",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "tony",
        },
        "ref": null,
        "rendered": "tony",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "path to pic",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            body of comment
          </p>,
          <p>
            tony
          </p>,
          <img
            src="path to pic"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "body of comment",
          },
          "ref": null,
          "rendered": "body of comment",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "tony",
          },
          "ref": null,
          "rendered": "tony",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "path to pic",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
